dashboard/src/main/home/app-dashboard/validate-apply/helm/HelmOverrides.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useMemo, useState } from "react";
+import styled from "styled-components";
+
+import api from "shared/api";
+
+import TabSelector from "components/TabSelector";
+import SelectRow from "components/form-components/SelectRow";
+import { MetricNormalizer, resolutions, secondsBeforeNow } from "../../expanded-app/metrics/utils";
+import { Metric, MetricType, NginxStatusMetric } from "../../expanded-app/metrics/types";
+import { match } from "ts-pattern";
+import { AvailableMetrics, NormalizedMetricsData } from "main/home/cluster-dashboard/expanded-chart/metrics/types";
+import MetricsChart from "../../expanded-app/metrics/MetricsChart";
+import { useQuery } from "@tanstack/react-query";
+import Loading from "components/Loading";
+import CheckboxRow from "components/CheckboxRow";
+import { PorterApp } from "@porter-dev/api-contracts";
+import { useLocation } from "react-router";
+import yaml from "js-yaml";
+import YamlEditor from "../../../../../components/YamlEditor";
+import Spacer from "../../../../../components/porter/Spacer";
+import Text from "../../../../../components/porter/Text";
+import Button from "../../../../../components/porter/Button";
+import {useFormContext} from "react-hook-form";
+import {PorterAppFormData} from "../../../../../lib/porter-apps";
+
+type PropsType = {
+  projectId: number;
+  clusterId: number;
+  appName: string;
+  appId: number;
+  deploymentTargetId: string;
+  overrideValues: any;
+  setError: (errorString: string) => void;
+};
+
+const HelmOverrides: React.FunctionComponent<PropsType> = ({
+  overrideValues,
+    setError,
+}) => {
+
+  const { setValue } = useFormContext<PorterAppFormData>();
+  const [currentOverrideValues, setOverrideValues] = React.useState<string>(overrideValues);
+
+  const setFormValue = (value: string) => {
+        setOverrideValues(value);
+        try {
+            if (value == "" || value == null || value == undefined) {
+                setValue("app.helmOverrides", "");
+            } else {
+                const jsonValues = yaml.load(value);
+                setValue("app.helmOverrides", JSON.stringify(jsonValues));
+            }
+            setError("");
```

Comment:
value will never be null or undefined here
```suggestion
           const newJsonValues = value ? yaml.load(value) : ""
           setValue("app.helmOverrides", JSON.stringify(newJsonValues))
            setError("");
```